linker/descriptors.go

```diff
@@ -272,28 +273,18 @@ func (r *result) Services() protoreflect.ServiceDescriptors {
 }
 
 func (r *result) PopulateSourceCodeInfo() {
-	srcLocProtos := asSourceLocations(r.FileDescriptorProto().GetSourceCodeInfo().GetLocation())
-	srcLocIndex := computeSourceLocIndex(srcLocProtos)
+	srcLocProtos, srcLocIndex := asSourceLocations(r.FileDescriptorProto().GetSourceCodeInfo().GetLocation())
 	r.srcLocations = srcLocs{file: r, locs: srcLocProtos, index: srcLocIndex}
 }
 
 func (r *result) SourceLocations() protoreflect.SourceLocations {
 	return &r.srcLocations
 }
 
-func computeSourceLocIndex(locs []protoreflect.SourceLocation) map[sourcePathKey]int {
-	index := map[sourcePathKey]int{}
-	for i, loc := range locs {
-		if loc.Next == 0 {
-			index[pathKey(loc.Path)] = i
-		}
```

Comment:
Previously, it looks like we indexed the _last_ location for a given path (`loc.Next == 0`).

The new code indexes the first one, which I believe is what protobuf-go does and is much more useful since the caller can then use `Next` in the returned location to traverse other locations for the same path.